pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -288,6 +289,9 @@ func translateRoute(in *networking.HTTPRoute,
 			}}
 	} else {
 		d := util.GogoDurationToDuration(in.Timeout)
+		if d == 0 {
+			d = 5 * time.Second
+		}
```

Comment:
please remove.. if timeout is omitted, the request will have no timeout (i.e. it can wait as long as it needs)

Reply:
FWIW we almost always regret requests without a timeout. IMO setting some timeout is the correct default behavior, though I'd probably make it more generous (say, 60s). Really it's just about ensuring that we'll eventually free resources server side even if the client doesn't behave well.

Reply:
I think I added this because envoy was crashing if time was 0 - will double check.